code/game/machinery/computer/security.dm

```diff
@@ -1,46 +1,321 @@
+			if (!sanitised_name || !sanity_check(params["ref"]))
+				return
+
+			message_admins("[key_name(ui.user)] has changed the record name of [active_record_general.fields["name"]] to [sanitised_name]")
+			active_record_general.fields["name"] = sanitised_name
+
+		if("toggle_sex")
+			if (!sanity_check(params["ref"]))
+				return
+
+			if (active_record_general.fields["sex"] == "male")
+				active_record_general.fields["sex"] = "female"
+			else
+				active_record_general.fields["sex"] = "male"
+
+		if("edit_age")
+			if (!sanity_check(params["ref"]))
+				return
+
+			// New records have a string in their age field which will break this input.
+			var/default_age = active_record_general.fields["age"] == "Unknown" ? 0 : active_record_general.fields["age"]
+
+			var/input = tgui_input_number(ui.user, "Enter a new age", "New Age", default_age, AGE_MAX, AGE_MIN)
+			if (!input || !sanity_check(params["ref"]))
+				return
+
+			active_record_general.fields["age"] = input
+
+		if ("set_criminal_status")
+			if (!istype(active_record_security, /datum/data/record) || !sanity_check(params["ref"]))
+				return
+
+			var/new_wanted_status = params["new_status"]
+			if(!new_wanted_status || !(new_wanted_status in WANTED_STATUSES))
+				return
+
+			active_record_security.fields["criminal"] = new_wanted_status
+
+			var/mob_gid = hex2num(active_record_general.fields["id"])
+			for(var/mob/living/carbon/human/human as anything in GLOB.human_mob_list)
+				if (human.gid == mob_gid)
+					human.sec_hud_set_security_status()
+					break
+
+		if ("add_comment")
+			if (!istype(active_record_security, /datum/data/record) || !sanity_check(params["ref"]))
+				return
+
+			var/input = tgui_input_text(ui.user, "Your name and time will be added to this new comment", "New Comment", multiline = TRUE)
+			if (!input || !istype(active_record_security, /datum/data/record) || !sanity_check(params["ref"]))
+				return
+
+			var/created_at = text("[]&nbsp;&nbsp;[]&nbsp;&nbsp;[]", time2text(world.realtime, "MMM DD"), time2text(world.time, "[worldtime2text()]:ss"), game_year)
+
+			var/created_by_name = ""
+			var/created_by_rank = ""
+			if(istype(ui.user, /mob/living/carbon/human))
+				var/mob/living/carbon/human/human = ui.user
+				created_by_name = human.get_authentification_name()
+				created_by_rank = human.get_assignment()
+			else if(istype(ui.user, /mob/living/silicon/robot))
+				var/mob/living/silicon/robot/robot = ui.user
+				created_by_name = robot.name
+				created_by_rank = "[robot.modtype] [robot.braintype]"
+
+			var/new_comment = list(
+				"entry" = input,
+				"created_by_name" = created_by_name,
+				"created_by_rank" = created_by_rank,
+				"created_at" = created_at,
+				"deleted_by" = null,
+				"deleted_at" = null
+			)
+
+			active_record_security.fields["comments"] += list(new_comment)
+
+			to_chat(ui.user, text("You have added a new comment to the Security Record of [].", active_record_security.fields["name"]))
```

Comment:
Do we want to log who changes the comments on a sec record? It might be belt and braces but it would help track down a griefer